### ISSUE TITLE
Ansible.ModuleUtils.Legacy Fail-Json is failing when being used with IIS based windows modules

### ISSUE THREAD

**Issue by carpnick** [ansible/ansible#32696]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
win_iis_*

##### ANSIBLE VERSION
```
ansible 2.4.1.0
  config file = /Users/ncarpenter/code/github/AWS_playbooks/ansible.cfg
  configured module search path = [u'/Users/ncarpenter/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 2.7.12 (default, Oct 11 2016, 05:16:02) [GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]

```

##### CONFIGURATION
```
DEFAULT_HOST_LIST(/Users/ncarpenter/code/github/AWS_playbooks/ansible.cfg) = [u'/Users/ncarpenter/code/github/AWS_playbooks/environments/dev']
```

##### OS / ENVIRONMENT
MAC OSX Sierra 10.12.4 - Control node
Windows 2012  - Attempting to control

##### SUMMARY
Unhandled exceptions are being propagated from Ansible.ModuleUtils.Legacy Fail-Json method when being used by several IIS based modules:

Lines that are hitting the issue(a few examples, it is throughout the iis modules):

1.  win_iis_virtualdirectory:
    - [Link1](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_iis_virtualdirectory.ps1#L29)
    - [Link2](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_iis_virtualdirectory.ps1#L35)
    - [Link3](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_iis_virtualdirectory.ps1#L79)
2. win_iis_website
    - [Link1](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_iis_website.ps1#L28)
    - [Link2](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_iis_website.ps1#L35)

##### STEPS TO REPRODUCE

```yaml
- name: Create a virtual directory with missing parameter name
  win_iis_virtualdirectory:
    site: "Default Web Site"
    state: present
    physical_path: "C:\\"
  register: virt_dir_creation
```

##### EXPECTED RESULTS
Error should be created that hits this [line](https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_iis_virtualdirectory.ps1#L29)

So it should say it failed due to  a missing required argument.

##### ACTUAL RESULTS

```
EXEC (via pipeline wrapper)
fatal: [10.100.28.119]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "module_stderr": "Method invocation failed because [System.Management.Automation.PSCustomObject] \r\ndoes not contain a method named 'ContainsKey'.\r\nAt line:95 char:9\r\n+     if (-not $obj.ContainsKey('changed')) {\r\n+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : InvalidOperation: (ContainsKey:String) [], Paren \r\n   tContainsErrorRecordException\r\n    + FullyQualifiedErrorId : MethodNotFound\r\n \r\n\r\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "rc": 1
}
fatal: [10.100.29.62]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "module_stderr": "Method invocation failed because [System.Management.Automation.PSCustomObject] \r\ndoes not contain a method named 'ContainsKey'.\r\nAt line:95 char:9\r\n+     if (-not $obj.ContainsKey('changed')) {\r\n+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\r\n    + CategoryInfo          : InvalidOperation: (ContainsKey:String) [], Paren \r\n   tContainsErrorRecordException\r\n    + FullyQualifiedErrorId : MethodNotFound\r\n \r\n\r\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "rc": 1
}
```

##### Other Notes Probably Secondary bug
[This Code](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Legacy.psm1#L78) is incorrectly assuming that a psobject has the method [ContainsKey](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/powershell/Ansible.ModuleUtils.Legacy.psm1#L95) on it.  This might be a separate bug.  The [PSObject class](https://docs.microsoft.com/en-us/dotnet/api/system.management.automation.psobject?view=powershellsdk-1.1.0) does not contain that method on it.

**Comment by carpnick** [comment]
@crazycoop74 FYI.

**Comment by jborean93** [comment]
Thanks for picking this up @carpnick, are you able to test out PR https://github.com/ansible/ansible/pull/32710 and give some info on that page to see if it fixes your issue?

**Comment by jborean93** [comment]
Fix was cherry-picked to stable 2.4 https://github.com/ansible/ansible/commit/b059dfe69f94333f22b439e67ba3b7ab56b3b24b, will be released with 2.4.2rc1
